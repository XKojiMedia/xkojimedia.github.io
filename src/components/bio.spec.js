import React from "react"
import renderer from "react-test-renderer"
import { useStaticQuery } from 'gatsby';

import Bio from "./bio";

describe("Header", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it("renders correctly", () => {
    useStaticQuery.mockReturnValue({
      avatar: {
        childImageSharp: {
          fixed: {
            width: 10,
            height: 10,
            src: '',
            srcSet: '',
          },
        }
      },
      site: {
        siteMetadata: {
          author: {
            name: '',
            summary: '',
          },
          social: {
            twitter: '@handle',
          }
        }
      }
    });
    const tree = renderer
      .create(<Bio />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
