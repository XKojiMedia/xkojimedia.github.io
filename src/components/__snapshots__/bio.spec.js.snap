// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "marginBottom": "4.375rem",
    }
  }
>
  <div
    className=" gatsby-image-wrapper"
    style={
      Object {
        "borderRadius": "100%",
        "display": "inline-block",
        "height": 10,
        "marginBottom": 0,
        "marginRight": "0.875rem",
        "minWidth": 50,
        "overflow": "hidden",
        "position": "relative",
        "width": 10,
      }
    }
  >
    <picture>
      <source
        srcSet=""
      />
      <img
        alt=""
        height={10}
        loading="lazy"
        onLoad={[Function]}
        src=""
        srcSet=""
        style={
          Object {
            "borderRadius": "50%",
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 500ms",
            "width": "100%",
          }
        }
        width={10}
      />
    </picture>
    <noscript
      dangerouslySetInnerHTML={
        Object {
          "__html": "<picture><source srcset=\\"\\" /><img loading=\\"lazy\\" width=\\"10\\" height=\\"10\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
        }
      }
    />
  </div>
  <p>
    Written by 
    <strong>
      
    </strong>
     
    
     
    <a
      href="https://twitter.com/@handle"
    >
      You should follow him on Twitter
    </a>
  </p>
</div>
`;
